geode-core/src/main/kotlin/org/apache/geode/cache/rest/controllers/RegionsController.kt

```diff
@@ -0,0 +1,74 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.geode.cache.rest.controllers
+
+import org.apache.geode.cache.configuration.RegionAttributesDataPolicy
+import org.apache.geode.cache.configuration.RegionAttributesType
+import org.apache.geode.cache.configuration.RegionConfig
+import org.apache.geode.internal.cache.GemFireCacheImpl
+import org.apache.geode.management.internal.cli.CliUtil
+import org.apache.geode.management.internal.cli.functions.CliFunctionResult
+import org.apache.geode.management.internal.cli.functions.CreateRegionFunctionArgs
+import org.apache.geode.management.internal.cli.functions.RegionCreateFunction
+import org.springframework.http.HttpStatus
+import org.springframework.http.ResponseEntity
+import org.springframework.web.bind.annotation.*
+
+data class RegionCreateParams @JvmOverloads constructor(val name: String = "", val type: String = "")
+
+@RestController
+class RegionsController {
+
+    @RequestMapping(method = [RequestMethod.POST], value = ["/regions"],
+            produces = ["application/json"], consumes = ["application/json"])
+    fun create(@ModelAttribute params: RegionCreateParams): ResponseEntity<String> {
+        var regionConfig = RegionConfig()
+        regionConfig.name = params.name.split("/").last()
+
+        setAttributes(regionConfig, params.type)
+
+        var functionArgs = CreateRegionFunctionArgs(params.name, regionConfig, true)
```

Comment:
As a first step to get something running, it's reasonable to put lots of logic into a controller method. But very soon, you end up wanting your controller methods to be "skinny".

I've used Spring very little. But my take is that, as with e.g. Jersey, a Spring controller is very much concerned with the HTTP protocol. That being the case, it seems like you'd like all your logic in a place that has almost nothing to do with the HTTP protocol. The benefit is that:
1. you can test that logic with plain old Java code—no need for an HTTP client
2. you can reuse that logic in non-HTTP contexts (other protocols, other situations)

Reply:
Yes, definitely would be moving a lot of this stuff out of controller code